projects/ecodefi/index.js

```diff
@@ -0,0 +1,177 @@
+
+  return stakedValue;
+}
+
+async function getTVLByTotalSupply(chain, token, block) {
+  let balances = {}
+  let tvl = BigNumber("0");
+
+  let { output: asset} = await sdk.api.abi.call({
+    block,
+    target: token,
+    abi: eTokenAbi['underlying'],
+    chain: chain
+  });
+
+  let { output: decimals} = await sdk.api.abi.call({
+    block,
+    target: asset,
+    abi: 'erc20:decimals',
+    chain: chain
+  });
+
+  let { output: assetTotalSupply } = await sdk.api.abi.call({
+    block,
+    target: token,
+    abi: eTokenAbi['totalSupply'],
+    chain: chain
+  });
+
+  let { output: exchangeRate} = await sdk.api.abi.call({
+    block,
+    target: token,
+    abi: eTokenAbi['exchangeRateCurrent'],
+    chain: chain
+  });
+
+  let assetPrice = await getUnderlyingPrice(chain, token, block);
+
+  tvl = tvl.plus(BigNumber(assetTotalSupply).times(BigNumber(exchangeRate)).times(BigNumber(assetPrice)).div(Double).div(BASE));
+  let convertedBalance = BigNumber(assetTotalSupply);
+
+  return {
+    convertedBalance,
+    tvl
+  }
+}
+
+async function getAllMarketsByChain(chain, block) {
+  const markets = await Promise.all(
+    allControllers[chain].map(async controller => {return (await
+      sdk.api.abi.call({
+        block,
+        target: controller,
+        abi: abi['getAllMarkets'],
+        chain: chain
+      })).output;
+    }));
+
+  return markets.flat();
+}
+
+async function getUnderlyingPrice(chain, token, block) {
+  const { output: eTokenPrices }  = await sdk.api.abi.call({
+    block,
+    target: oracles[chain],
+    params: token,
+    abi: oracleAbi['getUnderlyingPrice'],
+    chain: chain
+  });
+
+  return eTokenPrices;
+}
+
+async function getTVLOfeToken(chain, block) {
+  let eTokenBalances = {};
+  let eTokenTVL = BigNumber("0");
+  let eTokens = await getAllMarketsByChain(chain, block);
+
+  await (
+    Promise.all(eTokens.map(async (eToken) => {
+      let {
+	      convertedBalance: convertedBalance,
+	      tvl: tvl 
+      } = await getTVLByTotalSupply(chain, eToken, block);
+      eTokenTVL = eTokenTVL.plus(tvl);
+    }))
+  );
+
+  return eTokenTVL
+}
+
+async function getTVLByChain(chain, block) {
+  let balances = {};
+  let tvl = BigNumber("0");
+  tvl = tvl.plus(await getESGStakingValue(chain, block));
+
+  let eTokenTVL = await getTVLOfeToken(chain, block);
+  tvl = tvl.plus(BigNumber(eTokenTVL));
+  return tvl.toNumber();
+}
+
+async function bsc(timestamp, ethBlock, chainBlocks) {
+  let tvl = await getTVLByChain('bsc', chainBlocks['bsc']);
+  return toUSDTBalances(tvl);
+}
+
+async function staking(timestamp, ethBlock, chainBlocks) {
+  let staked = await getESGStakingValue('bsc', chainBlocks['bsc']);
+  return toUSDTBalances(staked);
+}
+
+module.exports = {
+  timetravel: true,
+  misrepresentedTokens: false,
+  methodology: 'counts the number of Total value locked in ESG protocol.',
+  bsc: {
+	  tvl: bsc,
```

Comment:
the staking part looks good, can you update the tvl computation part ot below, we have a helper that makes it easy to compute both deposits and borrowed amount:

    const { compoundExports } = require('../helper/compound');


    module.exports = {
      bsc: {
        ...compoundExports('0xfd1f241ba25b8966a14865cb22a4ea3d24c92451', 'bsc'),
        staking,
      },
    };